File: Programmers/삼각 달팽이_68625.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int n) {
    vector<int> answer;
    vector<vector<int>> board(n, vector<int>(n, 0));
    
    int num = 1;    // 넣을 수
    int y = -1, x = 0;
    
    for(int rnd=0; rnd<n;rnd++){  // round: 회차
        for (int j=rnd; j<n; j++){
            if (rnd%3 ==0)
                y++;
            else if (rnd%3==1)
                x++;
            else {x--; y--;}
            board[y][x] = num++;
        }
    }
    for (int i=0;i<n; i++)
        for(int j=0; j<i+1; j++)
            answer.push_back(board[i][j]);
    
    return answer;
}

//https://programmers.co.kr/learn/courses/30/lessons/68645

File: Programmers/L2/삼각 달팽이_68625.py
def solution(n):
    answer = [['0' for _ in range(i)] for i in range(1, n+1)]
    num = 1
    y, x = -1, 0
    for cnt in range(n):
        for _ in range(cnt, n):
            if cnt%3 == 0:    # 왼쪽 아래로 내려가기
                y += 1
            elif cnt%3 == 1:    # 오른쪽으로 이동하기
                x += 1
            else:   # 상한 대각선으로 이동하기
                y -=1
                x -=1
            answer[y][x] = num
            num += 1
            
    answer = sum(answer, [])

    return answer
  
'''
rnd : round로 회차.
        
https://programmers.co.kr/learn/courses/30/lessons/68645
'''
